include/bluetooth/bluetooth.h

```diff
@@ -297,21 +342,49 @@ enum {
 
 	/** Use whitelist to filter devices that can connect. */
 	BT_LE_ADV_OPT_FILTER_CONN = BIT(7),
+
+	/** Advertise with extended data length. */
+	BT_LE_ADV_OPT_EXT_DATA = BIT(8),
+
+	/* Advertise without a device address (identity or RPA). */
+	BT_LE_ADV_OPT_ANONYMOUS = BIT(9),
+
+	/* Advertise with transmit power. */
+	BT_LE_ADV_OPT_USE_TX_POWER = BIT(10),
+
+	/* Generate scan reports when advertiser was scanned by active scanner.
+	 */
+	BT_LE_ADV_OPT_SCAN_REPORTS = BIT(11),
+
+	/* Primary Advertising on coded PHY */
```

Comment:
Yeah, I think that suggestion makes a lot of sense.

So since you are saying secondary PHY is 2M then we also make the assumptions that most of the scanners support 2M secondary scanning. And only the rare cases only have 1M?

I would then make this suggestion: 
Default 2M
```
BT_LE_ADV_OPT_EXT_DATA       // 1M -> 2M
BT_LE_ADV_OPT_EXT_DATA_NO_2M // 1M -> 1M
BT_LE_ADV_OPT_LONG_RANGE     // coded -> coded
```
Or alternatively user choses 1M or 2M
```
BT_LE_ADV_OPT_EXT_DATA_1M  // 1M -> 1M
BT_LE_ADV_OPT_EXT_DATA_2M  // 1M -> 2M
BT_LE_ADV_OPT_LONG_RANGE   // coded -> coded
```

Reply:
Having device capeable of scanning for AE, and not being able to receive 2M packets is likely very rare. I have a slight preference for suggestion 1 as it will default sending on 1M/2M which is more optimal in terms of power consumption

Reply:
>BT_LE_ADV_OPT_LONG_RANGE

I completely disagree with using `LONG_RANGE` in the API for the following reasons:
- It is not in the spec at all (except when listing the features added in a particular spec version
- Almost all API users end up needing to check with the spec, and they won't find the mapping between this and coded phy
- Coded phy has potential uses beyond long range (better noise resilience)
- All  sniffer and HCI logs will use CODED instead

Reply:
```
BT_LE_ADV_OPT_EXT_DATA       // 1M -> 2M
BT_LE_ADV_OPT_EXT_DATA_NO_2M // 1M -> 1M
BT_LE_ADV_OPT_CODED     // coded -> coded
```
is reasonable, but do we need an `BT_LE_ADV_OPT_EXT_DATA_CODED`?

Reply:
@carlescufi . The use case of `BT_LE_ADV_OPT_EXT_DATA_LONG_RANGE` is long range establlishement. This seems like a use case that we should support

Reply:
@rugeGerritsen I may have missed something here. The only thing I'm advocating for is using `_CODED_` instead of `_LONG_RANGE_` and perhaps *add* a 4th macro `BT_LE_ADV_OPT_EXT_DATA_CODED`. I now understand that coded is always ext data so in fact the only change I am proposing is really renaming `BT_LE_ADV_OPT_LONG_RANGE` with `BT_LE_ADV_OPT_CODED` and then ask users to pass (BT_LE_ADV_OPT_EXT_DATA |  BT_LE_ADV_OPT_CODED)